http_handler_test.go

```diff
@@ -120,18 +118,18 @@ func (suite *HttpHandlerTestSuite) TestPutHandler_InvalidPredicate() {
 	suite.annotationsService.On("Write", knownUUID, annotationLifecycle, platformVersion, suite.tid, suite.annotations).Return(annotations.UnsupportedPredicateErr)
 	request := newRequest("PUT", fmt.Sprintf("/content/%s/annotations/%s", knownUUID, annotationLifecycle), "application/json", suite.body)
 	request.Header.Add("X-Request-Id", "tid_sample")
-	handler := httpHandler{suite.annotationsService, suite.producer, suite.originMap, suite.lifecycleMap, suite.messageType, suite.log}
+	handler := httpHandler{suite.annotationsService, suite.forwarder, suite.originMap, suite.lifecycleMap, suite.messageType, suite.log}
 	rec := httptest.NewRecorder()
 	router(&handler, &suite.healthCheckHandler, suite.log).ServeHTTP(rec, request)
 	assert.True(suite.T(), http.StatusBadRequest == rec.Code, fmt.Sprintf("Wrong response code, was %d, should be %d", rec.Code, http.StatusBadRequest))
 }
 
 func (suite *HttpHandlerTestSuite) TestPutHandler_ForwardingFailed() {
 	suite.annotationsService.On("Write", knownUUID, annotationLifecycle, platformVersion, suite.tid, suite.annotations).Return(nil)
-	suite.producer.On("SendMessage", mock.Anything).Return(errors.New("Forwarding failed"))
+	suite.forwarder.On("SendMessage", mock.Anything, mock.Anything, mock.Anything, mock.Anything).Return(errors.New("forwarding failed"))
```

Comment:
We could improve the test by checking for specific values.
```suggestion
	suite.forwarder.On("SendMessage", suite.tid, mock.AnythingOfType("string"), knownUUID, suite.annotations).Return(errors.New("forwarding failed"))
```

Reply:
I don't see an advantage in being more specific, we are checking the error here.

Reply:
We are checking both that `SendMessage` is called, and that the produced error is handled correctly. For me it makes sense to test for the parameters of `SendMessage`. 
Besides currently there is no explicit test in `http_handlers_test` that we provide `SendMessage` with the right values. So probably we should change the mock in `TestPutHandler_Success` as well.